x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutAutoFollowPatternAction.java

```diff
@@ -196,6 +200,12 @@ private static void markExistingIndicesAsAutoFollowed(
         for (final IndexMetadata indexMetadata : leaderMetadata) {
             IndexAbstraction indexAbstraction = leaderMetadata.getIndicesLookup().get(indexMetadata.getIndex().getName());
             if (AutoFollowPattern.match(patterns, indexAbstraction)) {
+                if (indexAbstraction.isSystem()) {
```

Comment:
I am torn on this. I think the logic here ensures that we do not follow these existing system indices anyway and therefore it hardly makes sense to have the deprecation warning here? I would opt to remove it.

Reply:
My thinking here was that we'll deprecate this behaviour in `7.14` but keep auto following system indices until `8.0`. is that a misunderstanding on my side?